lib/engine-jit/src/artifact.rs

```diff
@@ -325,15 +327,32 @@ impl Artifact for JITArtifact {
         &self.func_data_registry
     }
     fn serialize(&self) -> Result<Vec<u8>, SerializeError> {
-        // let mut s = flexbuffers::FlexbufferSerializer::new();
-        // self.serializable.serialize(&mut s).map_err(|e| SerializeError::Generic(format!("{:?}", e)));
-        // Ok(s.take_buffer())
-        let bytes = bincode::serialize(&self.serializable)
-            .map_err(|e| SerializeError::Generic(format!("{:?}", e)))?;
-
         // Prepend the header.
         let mut serialized = Self::MAGIC_HEADER.to_vec();
-        serialized.extend(bytes);
+
+        let mut metadata_binary = vec![0; 12];
+        let mut writable = &mut metadata_binary[..];
+        let serialized_data = self.serializable.serialize()?;
+        let length = serialized_data.len();
+        leb128::write::unsigned(&mut writable, length as u64).expect("Should write number");
+
+        serialized.extend(metadata_binary);
+
+        let align = std::mem::align_of::<SerializableModule>() as u64;
+        // println!("[SER] Metadata: First 16 bytes {:?}.\nLast 16 bytes: {:?}\nTotal length: {}", &serialized_data[..16], &serialized_data[length-16..], length);
+        let (serialized, offset) = append_data(serialized, &serialized_data, align);
+        // println!("[SER] First 50 bytes {:?}.\nAlign: {} - Offset: {}\nLast 50 bytes: {:?}", &serialized[..50], align, offset, &serialized[serialized.len()-50..]);
         Ok(serialized)
     }
 }
+
+pub fn append_data(mut prev_data: Vec<u8>, data: &[u8], align: u64) -> (Vec<u8>, u64) {
+    let align = align as usize;
```

Comment:
This code requires that align be a power of two, please assert that.

Reply:
Fixed